### PR TITLE
Uf 4082 create service layer

### PR DESCRIPTION
**Pull request by foobar-1337**

### CODE REVIEW

File: src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	 * 2. Enrich orders with data from SMEX.
+	 * 3. Store all orders in DB with status UNPROCESSED.
+	 * 4. Fetch all orders from DB with status FAILED and UNPROCESSED.
+	 * 5. Send orders to BillingPreProcessor and set status to PROCESSED/FAILED.
+	 * 6. If there are failed orders in database. Send mail to configured recipient.
+	 *
+	 * @param fromDate poll fromDate.
+	 * @param toDate   poll toDate.
+	 */
+	public void execute(final LocalDate fromDate, final LocalDate toDate) {
```
[the remark this answers is not in the document]
Comment (foobar-1337) [reply src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java:80]: The purpose with this method is to let the user execute the polling manually (e.g. re-runs when something has failed).
When executing the method manually I imagine that the user expects feedback on any errors that is thrown? If so: sync is preferred here.

The overall execution time is fast. In test environment 10 records was processed in ~ 1 second. 

TL;DR: I don't see why async would be the better choice here?

File: src/main/java/se/sundsvall/billingdatapolling/integration/messaging/configuration/MessagingConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+@Import(FeignConfiguration.class)
+public class MessagingConfiguration {
+
+    public static final String CLIENT_ID = "messaging";
+
+    @Bean
+	FeignBuilderCustomizer feignBuilderCustomizer(MessagingProperties messagingProperties, ClientRegistrationRepository clientRegistrationRepository) {
+        return FeignMultiCustomizer.create()
+			.withErrorDecoder(new ProblemErrorDecoder(CLIENT_ID))
```
[the remark this answers is not in the document]
Comment (foobar-1337) [reply src/main/java/se/sundsvall/billingdatapolling/integration/messaging/configuration/MessagingConfiguration.java:20]: It was one tab. The problem was that spaces where used on the other places (file copied from another repo). Now tabs are used everywhere.

**Comment by foobar-1337** [comment]
Acted on review comments

File: src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (josc04) [reply src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java:80]: Async might be a better choice if the execution time is longer than timeout setting for requests, but as you said - in test the exeution time is fast, so I guess it isn't a problem. 

I guess failures would still be sent as email to mailbox, so feedback on request isn't really needed right? But, as you said, the execution time is not an issue, so keep it as is.

**Review by josc04 — APPROVED** [review]

File: src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (foobar-1337) [reply src/main/java/se/sundsvall/billingdatapolling/service/scheduler/accesscard/AccessCardSchedulerService.java:80]: If the execution time is longer than the browsers timeout-setting, you will have the same effect as if the method was ascync: No response, but execution will complete, nevertheless.

Mail will be sent if the integration to Messaging-service is working. The mail-utility is an extra help-feature. In case of service/wso2/k8s/network/firewall/whatever-failure no messages will end up in the mailbox.
